src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx

```diff
@@ -143,170 +155,170 @@ export function DiscoverSidebar({
           defaultMessage: 'Index and fields',
         })}
       >
-        <div className="dscSidebar__sectionStatic">
-          <DiscoverIndexPattern
-            selectedIndexPattern={selectedIndexPattern}
-            setIndexPattern={setIndexPattern}
-            indexPatternList={sortBy(indexPatternList, (o) => o.attributes.title)}
-          />
-          <div className="dscSidebar__item">
+        <EuiFlexGroup
+          gutterSize="none"
+          responsive={false}
+          className="dscSidebar__content"
+          direction="column"
+        >
+          <EuiFlexItem grow={false}>
+            {' '}
+            <DiscoverIndexPattern
+              selectedIndexPattern={selectedIndexPattern}
+              setIndexPattern={setIndexPattern}
+              indexPatternList={sortBy(indexPatternList, (o) => o.attributes.title)}
+            />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
             <form>
               <DiscoverFieldSearch
                 onChange={onChangeFieldSearch}
                 value={fieldFilterState.name}
                 types={fieldTypes}
               />
             </form>
-          </div>
-        </div>
-        <div className="sidebar-list dscSidebar__sectionScroll">
-          {fields.length > 0 && (
-            <>
-              <EuiTitle size="xxxs" id="selected_fields">
-                <h3>
-                  <FormattedMessage
-                    id="discover.fieldChooser.filter.selectedFieldsTitle"
-                    defaultMessage="Selected fields"
-                  />
-                </h3>
-              </EuiTitle>
-              <EuiSpacer size="xs" />
-              <ul
-                className="dscSidebarList dscFieldList--selected"
-                aria-labelledby="selected_fields"
-                data-test-subj={`fieldList-selected`}
-              >
-                {selectedFields.map((field: IndexPatternField) => {
-                  return (
-                    <li
-                      key={`field${field.name}`}
-                      data-attr-field={field.name}
-                      className="dscSidebar__item"
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <div className="sidebar-list dscSidebar__scrollSection">
+              <div className="dscSidebar__scrollSectionList">
+                {fields.length > 0 && (
+                  <>
+                    <EuiAccordion
+                      id="dscSelectedFields"
+                      initialIsOpen={true}
+                      buttonContent={
+                        <EuiText size="xs">
+                          <strong>
+                            <FormattedMessage
+                              id="discover.fieldChooser.filter.selectedFieldsTitle"
+                              defaultMessage="Selected fields"
+                            />
+                          </strong>
+                        </EuiText>
+                      }
```

Comment:
Heavy +1 for removing this legacy stuff and moving to `EuiAccordion`
I think badges like in Lens displaying how much fields are available / selected would be useful for the user:
<img width="381" alt="Bildschirmfoto 2020-09-29 um 12 47 59" src="https://user-images.githubusercontent.com/463851/94549346-5a455880-0252-11eb-80c6-2cdfd617ff7c.png">

Reply:
@kertal I added the badges, not sure how we would handle the case where there's **Filters by type** on? In Lens they show the badge with the accent color if there are filters on. Would we want to do something similar in Discover? if so, how do we implement that?

See
https://github.com/elastic/kibana/blob/9a9a8b0c32fa808b83f891a2cd7f98b80c2bd57b/x-pack/plugins/lens/public/indexpattern_datasource/fields_accordion.tsx#L105

Reply:
sounds good, I'll have a look at lens

Reply:
I think we should adapt the lens behavior 
easiest way to find out whether a filter is active or not would be using `lodash` `isEqual` like this:

`const filterChanged = isEqual(fieldFilterState, getDefaultFieldFilter());`